### ISSUE TITLE
Whether the epochs is set to 200 or 300，there will be overfitting

### ISSUE THREAD

**Issue by xuezu29** [Megvii-BaseDetection/YOLOX#626]
![image](https://user-images.githubusercontent.com/10649774/131606866-9827d66d-a0a9-4f39-8af4-78945ed68e98.png)
epochs set to 300:  when training to about 200 epochs, the map starts to drop.
epochs set to 200:  when training to about 120 epochs, the map starts to drop.
Any suggestions?
thx!

**Comment by Joker316701882** [comment]
You may increase this value to 0.1 or 0.2 and try again.
https://github.com/Megvii-BaseDetection/YOLOX/blob/69f1173a569f1e0cc41025146a32b5d5fa06dd26/yolox/exp/yolox_base.py#L58

**Comment by xuezu29** [comment]
@Joker316701882 thx！ I have set  self.min_lr_ratio = 0.1 according to your suggestion  in wechat  and trainning.

**Comment by xuezu29** [comment]
![image](https://user-images.githubusercontent.com/10649774/131768980-a265d66c-48f9-4c9b-b0f1-762be2b8e642.png)

Even though the training is not complete, the results are basically the same. 
 both epochs=300.
 yellow curve : self.min_lr_ratio = 0.1
 blue curve:self.min_lr_ratio = 0.05

**Comment by Joker316701882** [comment]
I think you should increase the min_lr_ratio again and set the max_epochs=150. I suggest you try:
self.max_epochs = 150
self.min_lr_ratio = 0.5

**Comment by xuezu29** [comment]
@Joker316701882  thx！ I'll try.

**Comment by PyNancy** [comment]
@xuezu29   hi ,  how your model after adjust max_epochs and min_lr_ratio,  is normal ?  
i adjust max_epochs and min_lr_ratio as @Joker316701882  said, but still fast convergence and drop after some epochs, like below  

![image](https://user-images.githubusercontent.com/43673811/131939698-22d1c9b3-bd48-4e20-8204-20aa1b3c1ee1.png)

and even convert to mutilstep lr schedule as yolov3 and yolov4， but still not available /(ㄒoㄒ)/~~7

**Comment by xuezu29** [comment]
@PyNancy @Joker316701882 It does work though it's not perfect.
![image](https://user-images.githubusercontent.com/10649774/131942380-abd40b0c-9fc8-4131-aa6f-da5ac28c4c37.png)

**Comment by Joker316701882** [comment]
@xuezu29 I think you have obtained a better model compared to the previous one. But I don't understand the meaning of "perfect", the mAP is not satisfactory?

**Comment by PyNancy** [comment]
@xuezu29  your num_classes is 1? i think model AP should be stable with a little jitter instead of  AUC drop, @Joker316701882  is right ?

**Comment by Joker316701882** [comment]
@PyNancy It depends. If overfitting happens, the performance will drop.

**Comment by xuezu29** [comment]
@Joker316701882 Yes, it is much better than the previous model, thx a lot!  It drops a lot at the end with no data augmentation.

**Comment by xuezu29** [comment]
@PyNancy my num_classes: 9

**Comment by Joker316701882** [comment]
@xuezu29 This is an interesting phenomenon. You may reduce the no_aug_epochs to 1 or 0. But anyway, congrats on getting a better model.

**Comment by Joker316701882** [comment]
@PyNancy In your case, I think you did not properly use this project. You may need to trying to find the misuse of our code first.

**Comment by xuezu29** [comment]
> @xuezu29 This is an interesting phenomenon. You may reduce the no_aug_epochs to 1 or 0. But anyway, congrats on getting a better model.

Sincere thanks!

**Comment by PyNancy** [comment]
> @PyNancy In your case, I think you did not properly use this project. You may need to trying to find the misuse of our code first.

my datasets is single class and small object(4 pixel ~ 50 pixel) and big image (1k), maybe due to my datasets is special, but loss seem normal, 
![image](https://user-images.githubusercontent.com/43673811/131944110-e3cb9e3e-20d0-4bf3-9592-3e62b6963ba6.png)
Anyway, thank your great project and answer in time

**Comment by Joker316701882** [comment]
@PyNancy Did you pull

**Comment by PyNancy** [comment]
> @PyNancy Did you pull our latest branch? We recently fixed a fatal bug for detecting small objects.

yes, i pull the latest branch, and fix a fatal bug that bbox w,h limit  and rectangle train

**Comment by hudaodao69** [comment]
> @xuezu29 hi , how your model after adjust max_epochs and min_lr_ratio, is normal ? i adjust max_epochs and min_lr_ratio as @Joker316701882 said, but still fast convergence and drop after some epochs, like below
> 
> ![image](https://user-images.githubusercontent.com/43673811/131939698-22d1c9b3-bd48-4e20-8204-20aa1b3c1ee1.png)
> 
> and even convert to mutilstep lr schedule as yolov3 and yolov4， but still not available /(ㄒoㄒ)/~~7

sorry, could you please tell me how to add the pictures of F1 ,precision and recall in the Tensorboard.
